i3pyblocks/blocks/dbus.py

```diff
@@ -247,22 +247,33 @@ def __init__(
             **kwargs,
         )
         self.format = format
+        self.metadata = {
+            "artist": "",
+            "title": "",
+            "track_number": "",
+        }
+
+    def update_properties(
+        self,
+        properties: Dict[str, Variant],
+    ) -> None:
+        if "Metadata" in properties:
+            metadata = properties["Metadata"].value
+            if "xesam:artist" in metadata:
+                self.metadata["artist"] = ", ".join(metadata["xesam:artist"].value)
+            if "xesam:title" in metadata:
+                self.metadata["title"] = metadata["xesam:title"].value
+            if "xesam:trackNumber" in metadata:
+                self.metadata["track_number"] = metadata["xesam:trackNumber"].value
```

Comment:
I am thinking here if we shouldn't reset metadata in case of an update. My fear is that some program update only `xesam:title` for example, and you would have half the old information and half new.

Reply:
if only `xesam:title` is in the `Metadata` this will only update the `title` as long as the other fields are not present in the `updated_properites`

Reply:
Yeah, but my point is:

- Program X updates all 3 information
- Program Y assumes, however it only updates title

Now you have `artist` and `track_number` from program X and `title` from program Y. Correct me if I am wrong.

Reply:
This should not be possible since the `bus_name = "org.mpris.MediaPlayer2.{player}"` specifies the program. So program Y should not be able to assume control.

Since the **MPRIS** standard does require programs that have multiple instances to append a unique postfix like `org.mpris.MediaPlayer2.firefox.instance7389` this would not even pose a problem with multiple instances of for example firefox.

The instance thing is a whole other beast though that I want to try to address in the future (since firefox always appends an instance, I need to manually put that into the config and restart i3 to get it working).

Reply:
Huh... I was remembering wrong them. I was sure that this module allowed to be subscribed to a generic name so you could use any program instead a specific one. But yeah, you're right looking at the rest of the code (it was quite sometime that I wrote this module :sweat_smile: ).

Anyway, my point stands, just substitute Program Y with X. I am imagining a bugged program only update the metadata that is available to them. So if they only have access to title, they will update only title.

Reply:
If this is impossible to happen, please say so. Otherwise the code looks fine.

Reply:
I can not say if this would be possible to happen. I have only rudimentary understanding of **D-Bus** and **MPRIS**. A solution might be to cross reference with the `invalidated_properties`, but I did so far not pay any mind to what is in there. I will look into that tomorrow.

The code should be better than the status quo though, since without this patch a program only updating `xesam:title`, but not the rest will raise a `KeyError` and break the block.

Reply:
Ok, going to merge them. Future improvements can be done in other PRs if its found that this partial update is a real problem.